## Table of Contents
* [Question 1](#Question1)
* [Question 2](#Question2)
* [Question 3](#Question3)

In [183]:
import numpy as np
import statistics as stats
import pandas as pd

In [184]:
df = pd.read_csv('student-data.csv')
df

Name  Math  Science  Reading  History
0    Adam  11.0      9.0     16.0     13.5
1    Anna  12.0     11.0      8.5     10.0
2   James  11.0     18.0     11.5      9.0
3   Corey   7.0     15.5     11.0     14.0
4  Alisha   9.5     12.0     10.5     14.0
5   Jodie  15.0     18.5      7.0     12.0
6    John  18.0     15.5     11.0      7.5

In [185]:
name = ['Adam', 'Anna', 'James', 'Corey', 'Alisa', 'Jodie', 'John']
math = [11, 12, 11, 7, 9.5, 15, 18]
science = [9, 11, 18, 15.5, 12, 18.5, 15.5]
reading = [16, 8.5, 11.5, 11, 10.5, 7, 11]
history = [13.5, 10, 9, 14, 14, 12, 7.5]

## Question 1: <a class="anchor" id="Question1"></a>

In [187]:
math_mean = np.mean(math)
science_mean = np.mean(science)
reading_mean = np.mean(reading)
history_mean = np.mean(history)
print(f"The means for each subject are:\nMath: {math_mean}\nScience: {science_mean}\nReading: {reading_mean}\nHistory: {history_mean}")

The means for each subject are:
Math: 11.928571428571429
Science: 14.214285714285714
Reading: 10.785714285714286
History: 11.428571428571429


In [188]:
median_math = np.median(math)
print(f"The median math grade is: {median_math}")

The median math grade is: 11.0


In [189]:
mode_history = stats.mode(history)
print(f"Mode for History: {mode_history}")

Mode for History: 14


## Question 2: <a class="anchor" id="Question2"></a>

In [191]:
corrcoef_math_science = np.corrcoef(math, science)[0, 1]
corrcoef_math_science 

0.2694940962141907

In [192]:
corrcoef_math_history = np.corrcoef(math, history)[0, 1]
corrcoef_math_history

-0.7183583464149538

In [193]:
corrcoef_science_reading = np.corrcoef(science, reading)[0, 1]
corrcoef_science_reading

-0.5276058297032529

In [194]:
corrcoef_science_history = np.corrcoef(science, history)[0, 1]
corrcoef_science_history

-0.35868815565295303

In [195]:
corrcoef_reading_history = np.corrcoef(reading, history)[0, 1]
corrcoef_reading_history

0.20679404264832346

### Answer
* Math and history have the strongest correlation coefficient at -0.718

## Question 3: <a class="anchor" id="Question3"></a>

In [198]:
import math
math_grades = [24, 5, 15, 60, 54, 82, 99, 80, 70, 98, 93, 60, 33, 22, 65, 61, 51, 58, 83, 86, 42, 67, 60]
def desc_stats():
    grades_mean = np.mean(math_grades)
    grades_median = np.median(math_grades)
    grades_mode = stats.mode(math_grades)
    grades_min = np.min(math_grades)
    grades_max = np.max(math_grades)
    grades_range = grades_max - grades_min
    grades_var = stats.variance(math_grades)
    grades_std_dev = math.sqrt(grades_var)
    print(f"Mean: {grades_mean}\nMedian: {grades_median}\nMode: {grades_mode}\nMin: {grades_min}\nMax: {grades_max}\nRange: {grades_range}\nVariance: {grades_var}\nStandard Deviation: {grades_std_dev}")

desc_stats()

Mean: 59.47826086956522
Median: 60.0
Mode: 60
Min: 5
Max: 99
Range: 94
Variance: 694.3517786561265
Standard Deviation: 26.350555566365703
